tests/test_smoke.py

```diff
@@ -291,6 +291,11 @@ def test_minimal(generic_cloud: str):
             f'sky logs {name} 4 --status',  # Ensure the job succeeded.
             f'sky exec {name} \'echo "$SKYPILOT_CLUSTER_INFO" | jq .cloud | grep -i {generic_cloud}\'',
             f'sky logs {name} 5 --status',  # Ensure the job succeeded.
+            # Test '-c' for exec
+            f'sky exec -c {name} echo',
+            f'sky logs {name} 6 --status',
+            f'sky exec -c {name} echo hi test',
+            f'sky logs {name} 7 | grep "hi test"',
```

Comment:
Maybe add some tricky tests too (can be unit tests?)?
```
sky exec {name} -c {name}
sky exec {name} -c {name} echo

sky exec {name} 
sky exec -c {name} 
```

Reply:
I also tested this which passed. Should they pass?:
```
sky exec {name}1111 -c {name}  # passes
sky exec {name} -c {name}1111  # did not pass (expected)
```

Reply:
I added those tests in the smoke test for now, as it requires connection to the cluster, which might need some more design for unit test.

> I also tested this which passed. Should they pass?:
> 
> ```
> sky exec {name}1111 -c {name}  # passes
> sky exec {name} -c {name}1111  # did not pass (expected)
> ```

I think the behavior is expected, because when the user explicitly specifies `-c`, it should be considered as the cluster name, and the other parts should be the commands to run.